mesa: 20.2.4 -> 20.3.1

primeos opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

primeos committed:
mesa: 20.2.4 -> 20.3.0

primeos committed:
mesa: Drop all old patches

There's no need to keep the expressions for old patches around.
Also: Apply the musl patch unconditionally to ensure that it doesn't
break after future updates.

SuperSandro2000 commented on pkgs/development/libraries/mesa/default.nix:
```diff
@@ -58,38 +58,7 @@ stdenv.mkDerivation {
-    ++ lib.optionals stdenv.hostPlatform.isMusl [
-      # Fix `-Werror=int-conversion` pthread warnings on musl.
-      # TODO: Remove when https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/6121 is merged and available
-      (fetchpatch {
```
Please remove fetchpatch from inputs.

alapshin commented:
Is there are any plans of migrating to LLVM 11 to enable support for new AMD cards
https://www.phoronix.com/scan.php?page=news_item&px=AMDGPU-LLVM-11-Navi-2-GFX1030 ?

primeos committed:
mesa: Switch from LLVM 9 to llvmPackages_latest

The radeonsi and radv (without ACO) drivers requires LLVM 11 for
GFX 10.3 [0],[1]. E.g. Arch Linux also compiles Mesa with LLVM 11 so
this should be fine.

[0]: https://docs.mesa3d.org/relnotes/20.2.0.html
[1]: https://github.com/NixOS/nixpkgs/pull/105817#issuecomment-738888102

primeos commented:
@alapshin I wasn't aware of that, thanks! IIRC upstream CI mainly uses LLVM 9 but Arch already uses LLVM 11 so I think it should be fine to switch to `llvmPackages_latest` to avoid such issues in the future.

vcunat commented:
Yes, I think mesa generally supports the currently newest LLVM and sometimes it implies additional features in drivers.  This PR seems a good moment when to switch.

jansol commented:
[20.3.1 is out](https://docs.mesa3d.org/relnotes/20.3.1.html)

primeos committed:
mesa: 20.3.0 -> 20.3.1

primeos commented:
Tested `r600` and `iris`.
